src/preamble.js

```diff
@@ -297,7 +297,8 @@ var HEAP,
 /** @type {Float64Array} */
   HEAPF64;
 
-function updateGlobalBufferViews() {
+function updateGlobalBufferAndViews(buf) {
+  Module['buffer'] = buffer = buf;
```

Comment:
why is this added? this increases code size unnecessarily I think.

Reply:
When I authored this change in our fork, the preamble code used to read
```js
function updateGlobalBuffer(buf) {
  Module['buffer'] = buffer = buf;
}

function updateGlobalBufferViews() {
  Module['HEAP8'] = HEAP8 = new Int8Array(buffer);
  Module['HEAP16'] = HEAP16 = new Int16Array(buffer);
  Module['HEAP32'] = HEAP32 = new Int32Array(buffer);
  Module['HEAPU8'] = HEAPU8 = new Uint8Array(buffer);
  Module['HEAPU16'] = HEAPU16 = new Uint16Array(buffer);
  Module['HEAPU32'] = HEAPU32 = new Uint32Array(buffer);
  Module['HEAPF32'] = HEAPF32 = new Float32Array(buffer);
  Module['HEAPF64'] = HEAPF64 = new Float64Array(buffer);
}
```
so I merged the two functions into one since the two actions always need to be done together.

However it looks like since then the repository has been refactored to remove `updateGlobalBuffer` function and I cannot find an assignment to `Module['buffer'] = buffer = buf;` anywhere after a memory resize occurs? I take this to mean that the field `Module['buffer']` has since been deprecated? I'll remove the assignment.